test/parallel/test-shadow-realm-gc-module.js

```diff
@@ -8,12 +8,16 @@
 
 const common = require('../common');
 const fixtures = require('../common/fixtures');
+const { promisify } = require('util');
 
+let wait = promisify(setTimeout);
 async function main() {
   const mod = fixtures.fileURL('es-module-shadow-realm', 'state-counter.mjs');
   for (let i = 0; i < 100; i++) {
     const realm = new ShadowRealm();
     await realm.importValue(mod, 'getCounter');
+    // Give the GC some breathing room.
+    await wait(10);
```

Comment:
It's not certain that this would be enough. Locally it fixes the flake for me though. Another possible solution is using `checkIfCollectable()` but I am not quite sure if that achieves the purpose (the realm could technically still leak after the ShadowRealm object goes away in the principal realm, and `checkIfCollectable()` won't be able to detect the native leaks).

Reply:
It's enough on my computer.

Reply:
I think it may be worthwhile to add a helper function in `test/common/gc` for GC tight looping, to address the flakiness of `parallel/test-shadow-realm-gc` as reported in https://github.com/nodejs/reliability/issues/718.

Reply:
Fixed that as well.